crates/ruff/src/lib.rs

```diff
@@ -321,7 +321,9 @@ pub fn check(args: CheckCommand, log_level: LogLevel) -> Result<ExitStatus> {
         printer_flags,
     );
 
-    let preview = overrides.preview.unwrap_or_default().is_enabled();
+    // the settings should already be combined with the CLI overrides at this point
+    // TODO(jane): let's make this `PreviewMode`
+    let preview = pyproject_config.settings.linter.preview.is_enabled();
```

Comment:
We should also link a TODO here to #8232 to make sure this references the _global_ preview setting not the linter specific one